ql/src/test/queries/clientpositive/filter_literals.q

```diff
@@ -0,0 +1,25 @@
+
```

Comment:
Can you also check that the semantics of non-boolean expressions in the `WHERE` clause are the same with and without CBO and add tests if necessary. This is to ensure that we are not introducing behavior changes by allowing this kind of queries to use CBO.

For instance are the following queries return the same results with and without CBO?
```sql
WHERE 'foo';WHERE 'true';WHERE 'false';
WHERE 1;WHERE 0;WHERE -1;
WHERE null;
```

Reply:
I have added more tests for these, and the results are same with and without CBO.

Reply:
Thanks for the extra tests. 

Can you please include also `WHERE 'true'` and `WHERE 'false'`? In principle the behavior should be the same with `WHERE 'foo'` but I just want to ensure that we don't have any special interpretation for 'true' and 'false' when they appear as `STRING` literals.

Reply:
Your intuition is on point!

For `WHERE 'false'` we get empty values because `UDFToBoolean` returns `false` for `'false'` from `PrimitiveObjectInspectorUtils#parseBoolean()`:
```
  /**
   * Parses a boolean from string
   *
   * Accepts "false","off","no","0" and "" as FALSE
   * All other values are interpreted as true.
   */
  public static boolean parseBoolean(byte[] arr, int st, int len)
```
Logs:
```
optimizer.ConstantPropagateProcFactory: org.apache.hadoop.hive.ql.udf.generic.GenericUDFBridge([Const string false])=false
```

I feel that this is fine since the idea is to be more compliant with sql: https://issues.apache.org/jira/browse/HIVE-15939, but let me know what you think!

Reply:
If the behavior with and without CBO is the same for this "special" literals then its fine with me. We can just add the tests to document the current situation and leave things as they are. The reason I am asking for all the tests is to avoid accidental changes in behavior now or in the future.

Reply:
Added the tests to filter_literals.q